### PR TITLE
Basic react setup

### PR DESCRIPTION
**Pull request by Jack-Music**
This pull request intends to configure the development enviroment for the virtual MMX project

### Goals
- Build basic enviroment for react development

### Steps / detail
- setup babel
- eslint for linting
- prettier for formatting
- gitignore template from gitignore.io

This is my first time setting up react from scratch, so please let me know about any improvement or advise

### CODE REVIEW

**Comment by mozi-h** [comment]
If you want this merged, mark it as ready for review again, @Jack-Music

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+charset = utf-8
+indent_style = tab
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = true
+end_of_line = lf
+
+[*.md]
+max_line_length = off
+trim_trailing_whitespace = false
```
Comment (mozi-h) [inline .editorconfig:14]: Is this for your editor? If so, add to .gitignore

**Comment by mozi-h** [comment]
We are creating this together in a meeting now, thanks anyways!

File: .editorconfig (hunk 1 above)
Comment (Jack-Music) [reply .editorconfig:14]: editorconfig is a very popular way to keep a consistent coding style across IDEs and editors and should not be in a gitignore as a lot of popular repositories show (e.g.  react-bootstrap)

**Comment by Jack-Music** [comment]
Your welcome, i just watched that you are creating the app with create-react-app. In this pull request i decided to try a manual setup because a lot of articels recommended doing the manual way rather than using create-react-app to gain full control.

**Comment by mozi-h** [comment]
Huh, does the manual setup include hot reloading?
If we find `create-react-app` to be too out-of-control with its automation, we cound always redo it as a manual one and copy over `src`.

**Comment by Jack-Music** [comment]
> 
> 
> Huh, does the manual setup include hot reloading?
> If we find `create-react-app` to be too out-of-control with its automation, we cound always redo it as a manual one and copy over `src`.

Of course, webpack supports hot reloading and it is configured in the package.json.
But of couse let's stick with `create-react-app` until we need more control. Although i would suggest to extend the pretttier config and eslint config a bit more to have a consistent code style and add husky for automatical formatting and linting before commit ([More details](https://create-react-app.dev/docs/setting-up-your-editor/)). In case we want to have more control i would suggest to do a manual setup or switch to [Neutrino](https://neutrinojs.org/) rather dann simply use `create-react-app eject`.

**Pull request closed without merging**